pkg/util/configspec.go

```diff
@@ -190,3 +191,35 @@ func EnsureMinHardwareVersionInConfigSpec(
 	}
 	configSpec.Version = configSpecHwVersion.String()
 }
+
+// SafeConfigSpecToString returns the string-ified version of the provided
+// ConfigSpec, first trying to use the special JSON encoder, then defaulting to
+// the normal JSON encoder.
+func SafeConfigSpecToString(
+	configSpec *vimtypes.VirtualMachineConfigSpec) (s string) {
+
+	if configSpec == nil {
+		return ""
```

Comment:
May not matter since this is just used for logging, but normal json.Marshal would return `"null"` in this case.

Reply:
Yep. Fair, I'll update it.